myst_nb/parser.py

```diff
@@ -112,6 +113,33 @@ def parse_block(src, start_line):
                 dup_ref["fixed"] = True
         return tokens
 
+    def parse_code_cell(cell, start_line):
```

Comment:
move this to a separate function, add documentation, add type annotations

Reply:
do y'all have type checking tests i could run?

i don't think that this function can be moved to a separate function. i am following the pattern used on parse_block and parse-code-cell is invoking parse-block.

Reply:
> do y'all have type checking tests i could run

There is nothing automated setup as yet, but it would be good to add mypy (#204)

> i don't think that this function can be moved to a separate function

The problem is this function is getting quite big and complex. It would be ideal if it could be split in some way

Reply:
i split the function out of scope as a private function.